lib/config/fileconf.go

```diff
@@ -992,6 +992,9 @@ func (t StaticToken) Parse() ([]types.ProvisionTokenV1, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	if roles.Include(types.RoleBot) {
+		return nil, trace.BadParameter("role %q is not allowed in static token configuration", types.RoleBot)
+	}
```

Comment:
Should we add a warning in v18 and do a breaking change for v19?
Or add a warning for the next v17 and v16 releases

Reply:
That seems reasonable - I can do a backport and switch the error out for a warning.

Reply:
What does `Bot` in a static token even do?

Reply:
If it never actually worked and all this change does is cause the user to see a better error message then we should ship it in v18.

Reply:
I think that's the plan, my question was more about potentially backporting the error, but if Teleport currently starts (albeit with a broken join token) then we shouldn't break that.

Reply:
So configuring this does actually allow Teleport to start, however, when you try to join the Bot, it produces `ERROR: unsupported join method "" for bot`.

 If we want to take a more conservative route of emitting a warning in v18 and then making this a breaking change in 19 - that's fine by me. I'm a little concerned that a warning log in the early stages of a Teleport startup is unlikely to be spotted though.